packages/store/src/database/backends/indexedDB.test.ts

```diff
@@ -0,0 +1,170 @@
+import { IDBDatabaseBackend } from "./indexedDB";
+import { EventID, Task, TaskID } from "../../core2";
+import { testTask } from "../__tests__/testTasks";
+import { DatabaseBackendEntityRecord } from "../backend";
+// @ts-ignore Looks like there is no @types for this library
+import FDBFactory from "fake-indexeddb/lib/FDBFactory";
+import { EntityID, EntityType } from "../../core2/entities/entityBase";
+
+let backend: IDBDatabaseBackend;
+
+beforeEach(() => {
+  indexedDB = new FDBFactory();
+  backend = new IDBDatabaseBackend(indexedDB);
+});
+
+function createTestTask({
+  id,
+  lastEventID,
+  dueTimestampMillis,
+}: {
+  id: string;
+  lastEventID: string;
+  dueTimestampMillis: number;
+}): DatabaseBackendEntityRecord<Task> {
+  // lazy deep clone
+  const newTask = JSON.parse(JSON.stringify(testTask)) as Task;
+  newTask.id = id as TaskID;
+  newTask.componentStates["a"].dueTimestampMillis = dueTimestampMillis;
+  return { entity: newTask, lastEventID: lastEventID as EventID };
+}
+
+const testTasks: DatabaseBackendEntityRecord<Task>[] = [
+  createTestTask({ id: "a", lastEventID: "x", dueTimestampMillis: 50 }),
+  createTestTask({ id: "b", lastEventID: "y", dueTimestampMillis: 100 }),
+  createTestTask({ id: "c", lastEventID: "z", dueTimestampMillis: 150 }),
+];
+
+test("round-trip entities", async () => {
+  await backend.putEntities(testTasks);
+
+  const result = await backend.getEntities(["a", "c", "z"] as TaskID[]);
+  expect(result).toEqual(
+    new Map([
+      ["a", testTasks[0]],
+      ["c", testTasks[2]],
+    ]),
+  );
+});
+
+describe("task components", () => {
+  test("created on insert", async () => {
+    await backend.putEntities(testTasks);
+
+    const results = await fetchAllRowsForTable("derived_taskComponents");
+    const filteredResults = results.filter(
+      (component: { taskID: string }) => component.taskID == "a",
+    );
```

Comment:
Decided to just do a rudimentary filter rather than finicking with querying indexedDB using indexes. If you think the later would be beneficial, I can go ahead and convert :)